src/controls/TextControl.tsx

```diff
@@ -44,7 +45,9 @@ export function TextControl({
     (uischema.options as TextControlOptions) ?? {}
   const formItemProps =
     "formItemProps" in uischema ? uischema.formItemProps : {}
-  const tooltip = options.tooltip ? options.tooltip : formItemProps?.tooltip
+  const { tooltip, ...formItemPropsWOTooltip } = formItemProps as FormItemProps
```

Comment:
Casting here actually makes it possible to create a runtime error if formItemProps is undefined. Instead of casting, you just need to handle the undefined case for formItemProps, e.g. ```const { tooltip, ...formItemPropsWOTooltip } = formItemProps ?? {}```